src/rnnlm/rnnlm-decodable-simple-looped.h

```diff
@@ -0,0 +1,188 @@
+                   "by the neural net.");
+    opts->Register("debug-computation", &debug_computation, "If true, turn on "
+                   "debug for the actual computation (very verbose!)");
+
+    // register the optimization options with the prefix "optimization".
+    ParseOptions optimization_opts("optimization", opts);
+    optimize_config.Register(&optimization_opts);
+
+    // register the compute options with the prefix "computation".
+    ParseOptions compute_opts("computation", opts);
+    compute_config.Register(&compute_opts);
+  }
+};
+
+
+/**
+   When you instantiate class DecodableNnetSimpleLooped, you should give it
+   a const reference to this class, that has been previously initialized.
+ */
+class DecodableRnnlmSimpleLoopedInfo  {
+ public:
+  DecodableRnnlmSimpleLoopedInfo(
+      const DecodableRnnlmSimpleLoopedComputationOptions &opts,
+      const kaldi::nnet3::Nnet &rnnlm,
+      const CuMatrix<BaseFloat> &word_embedding_mat);
+
+  void Init(const DecodableRnnlmSimpleLoopedComputationOptions &opts,
+            const kaldi::nnet3::Nnet &rnnlm,
+            const CuMatrix<BaseFloat> &word_embedding_mat);
+
+  const DecodableRnnlmSimpleLoopedComputationOptions &opts;
+
+  const kaldi::nnet3::Nnet &rnnlm;
+  const CuMatrix<BaseFloat> &word_embedding_mat;
+
+  // frames_left_context equals the model left context plus the value of the
+  // --extra-left-context-initial option.
+  int32 frames_left_context;
+  // frames_right_context is the same as the right-context of the model.
+  int32 frames_right_context;
+  // The frames_per_chunk equals the number of input frames we need for each
+  // chunk (except for the first chunk).
+  int32 frames_per_chunk;
+
+  // The output dimension of the nnet neural network (not the final output).
+  int32 nnet_output_dim;
+
+  // The 3 computation requests that are used to create the looped
+  // computation are stored in the class, as we need them to work out
+  // exactly shich iVectors are needed.
+  ComputationRequest request1, request2, request3;
+
+  // The compiled, 'looped' computation.
+  NnetComputation computation;
+};
+
+/*
+  This class handles the neural net computation; it's mostly accessed
+  via other wrapper classes.
+
+  It accept just input features */
+class DecodableRnnlmSimpleLooped {
+ public:
+  /**
+     This constructor takes features as input.
+     Note: it stores references to all arguments to the constructor, so don't
+     delete them till this goes out of scope.
+
+     @param [in] info   This helper class contains all the static pre-computed information
+                        this class needs, and contains a pointer to the neural net.
+     @param [in] feats  The input feature matrix.
+  */
+  DecodableRnnlmSimpleLooped(const DecodableRnnlmSimpleLoopedInfo &info);
+
+  // returns the number of frames of likelihoods.  The same as feats_.NumRows()
+  inline int32 NumFrames() const { return num_frames_; }
```

Comment:
I think it might be better to give this function (if it is even needed) a name that's more meaningful for the current context-- no need to stick to the DecodableInterface.  Reconsider the names of other member functions too.